packages/mds-stream/kafka/stream-producer.ts

```diff
@@ -64,9 +74,15 @@ export const KafkaStreamProducer = <TMessage>(
     write: async (message: TMessage[] | TMessage) => {
       if (isDefined(producer)) {
         const messages = (isArray(message) ? message : [message]).map(msg => {
-          return { value: JSON.stringify(msg) }
-        })
+          /**
+           * Gets the key to partition on (e.g. a device_id) if partitionKey is set in the options.
+           * By default, this uses a hash of the key to determine what partition it should land on.
+           * If no partitionKey is specified in the options, none will be used.
+           */
+          const getKey = () => (partitionKey ? { key: msg[partitionKey] } : {})
```

Comment:
I think this function should be declare outside of the `map` and accept `msg` as a parameter so it’s not declared for each message.